src/api/undeploy/helpers/undeploy-service-from-environment.js

```diff
@@ -0,0 +1,79 @@
+import { config } from '~/src/config/index.js'
+import { registerUndeployment } from '~/src/api/undeploy/helpers/register-undeployment.js'
+import { removeDeploymentFile } from '~/src/api/undeploy/helpers/remove-deployment-file.js'
+import { lookupTenantService } from '~/src/api/deploy/helpers/lookup-tenant-service.js'
+import { isFeatureEnabled } from '~/src/helpers/feature-toggle/is-feature-enabled.js'
+
+const deployFromFileEnvironments = config.get('deployFromFileEnvironments')
+
+/**
+ * @typedef {import("pino").Logger} Logger
+ * @param {string} imageName
+ * @param {string} environment
+ * @param {{id: string, displayName: string}} user
+ * @param {Logger} logger
+ */
+async function undeployServiceFromEnvironment(
+  imageName,
+  environment,
+  user,
+  logger
```

Comment:
`logger` appears to be going on some sort of long journey. Is this needed? can you just use `createLogger()` in the final destination?